napari/layers/utils/stack_utils.py

```diff
@@ -0,0 +1,171 @@
+    else:
+        colormap = iter(colormap)
+
+    if blending not in ['additive', 'translucent', 'opaque']:
+        blending = 'additive'
+
+    if scale is None:
+        scale = np.delete(stack.scale, axis)
+
+    if translate is None:
+        translate = np.delete(stack.translate, axis)
+
+    if contrast_limits is None:
+        contrast_limits = n_channels * [None]
+
+    if gamma is None:
+        gamma = n_channels * [1]
+
+    kwargs = {
+        'rgb': stack.rgb,
+        'blending': blending,
+        'interpolation': stack.interpolation,
+        'rendering': stack.rendering,
+        'iso_threshold': stack.iso_threshold,
+        'attenuation': stack.attenuation,
+        'metadata': stack.metadata,
+        'scale': scale,
+        'translate': translate,
+        'opacity': stack.opacity,
+        'visible': stack.visible,
+        'multiscale': stack.multiscale,
+    }
+
+    imagelist = list()
+
+    for i in range(n_channels):
+        layer_name = f'{name} layer {i}'
+        try:
+            color = next(colormap)
+        except IndexError:
+            color = 'gray'
+
+        kwargs['contrast_limits'] = contrast_limits[i]
+        kwargs['gamma'] = gamma[i]
+        kwargs['colormap'] = color
+
+        image = Image(np.take(data, i, axis=axis), name=layer_name, **kwargs)
+
+        imagelist.append(image)
+
+    return imagelist
+
+
+def images_to_stack(
+    images: List[Union[Image, Labels]],
+    axis: int = 0,
+    rgb: bool = None,
+    colormap: Union[str, Colormap] = None,
+    contrast_limits: List[int] = None,
+    gamma: float = 1,
+    interpolation: str = 'nearest',
+    rendering: str = 'mip',
+    iso_threshold: float = 0.5,
+    attenuation: float = 0.5,
+    name: str = None,
+    metadata: dict = None,
+    scale: Tuple[float] = None,
+    translate: Tuple[float] = None,
+    opacity: float = 1,
+    blending: str = 'translucent',
+    visible: bool = True,
+    multiscale: bool = None,
+) -> Image:
+    """Function to combine selected image layers in one layer
+
+    Parameters
+    ----------
+    viewer : napari.viewer.Viewer
+        The viewer with the selected image
+    axis : int
+        Index to to insert the new axis
+
+    Returns
+    -------
+    stack : napari.layers.Image
+        Combined image stack
+    """
+
+    kwargs = {
+        'rgb': rgb,
+        'colormap': colormap,
+        'blending': blending,
+        'contrast_limits': contrast_limits,
+        'gamma': gamma,
+        'interpolation': interpolation,
+        'rendering': rendering,
+        'iso_threshold': iso_threshold,
+        'attenuation': attenuation,
+        'name': name,
+        'metadata': metadata,
+        'scale': scale,
+        'translate': translate,
+        'opacity': opacity,
+        'visible': visible,
+        'multiscale': multiscale,
+    }
+    new_list = [image.data for image in images]
+    new_data = np.stack(new_list, axis=axis)
+    stack = Image(new_data, **kwargs)
```

Comment:
This one still needs some work. It's unclear what should be done with things like colormap and contrast limits, but scale and translate should certainly be pilfered from the top image and a scale of `1` inserted at the new axis.